core/src/main/java/org/apache/iceberg/rest/auth/AuthConfig.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.auth;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.iceberg.rest.ResourcePaths;
+import org.immutables.value.Value;
+
+@Value.Style(redactedMask = "****")
+@SuppressWarnings("ImmutablesStyle")
+@Value.Immutable
+public interface AuthConfig {
```

Comment:
can we add some javadoc about this interface on what's the responsibility of AuthConfig and how it can be used/implementated?

Also I am unclear on whether we can implement a custom AuthConfig to account for proprietary token generation which is not Oauth2

Reply:
> Also I am unclear on whether we can implement a custom AuthConfig to account for proprietary token generation which is not Oauth2

`AuthConfig` is directly related to `AuthSession` and controls the behavior of it, which in turn is OAuth2. Also it's an Immutable interface where the implementation will be generated from (rather than having to implement something by hand). 
I've added some javadoc to point out that the purpose of `AuthConfig` is to hold configuration options from `AuthSession`. Hopefully that makes it clearer.

Reply:
thank you @nastra , that helps. But i dont see the new commit about the javadoc, maybe you forgot to push to the branch?

Reply:
Yep I forgot to push, thanks